Doc/library/urllib.parse.rst

```diff
@@ -275,14 +281,39 @@ or on combining URL components into a URL string.
    states that these are equivalent).
 
 
-.. function:: urlsplit(urlstring, scheme='', allow_fragments=True)
+.. function:: urlsplit(url, scheme='', allow_fragments=True)
+
+   Similar to :func:`urlparse`, without splitting the URL parameters.
+
+   Parse a URL into five components, returning a 5-item :term:`named tuple`.
+   This corresponds to the general structure of a URL:
+   ``scheme://netloc/path?query#fragment``.
+
+   The delimiters as shown above are not part of the result, except for a
+   leading slash in the path component, which is retained if present.
+
+   Additionally, the netloc property is broken down into these additional
+   attributes added to the returned object: username, password, hostname,
+   and port.
+
+   % escapes are not decoded.
+
+   The *scheme* argument gives the default addressing scheme, to be
+   used only if the URL does not specify one.  It should be the same type
+   (text or bytes) as *url*, except that the default value ``''`` is
+   always allowed, and is automatically converted to ``b''`` if appropriate.
+
+   If the *allow_fragments* argument is false, fragment identifiers are not
+   recognized.  Instead, they are parsed as part of the path, parameters
+   or query component, and :attr:`fragment` is set to the empty string in
+   the return value.
 
-   This is similar to :func:`urlparse`, but does not split the params from the URL.
-   This should generally be used instead of :func:`urlparse` if the more recent URL
-   syntax allowing parameters to be applied to each segment of the *path* portion
-   of the URL (see :rfc:`2396`) is wanted.  A separate function is needed to
-   separate the path segments and parameters.  This function returns a 5-item
-   :term:`named tuple`::
+   This is similar to :func:`urlparse`, but does not split the params from the
+   URL. This should generally be used instead of :func:`urlparse` if the more
+   recent URL syntax allowing parameters to be applied to each segment of the
+   *path* portion of the URL (see :rfc:`2396`) is wanted.  A separate function
+   is needed to separate the path segments and parameters.
+   This function returns a 5-item :term:`named tuple`::
```

Comment:
The paragraph here feels a bit confusing to the reader. Could it be rephrased a little bit and an example given:

```
The function is similar to :func:`urlparse`, but does not split the parameters from the
URL. This is useful, if the given :attr:`url` follows the alternative URL syntax that allows parameters
per path segment (e.g. `http://hostname/path;arg=value/bar`, see :rfc:`2396`).
In that case, the path segments and parameters need to be separated with a separate function.

The function returns a 5-item :term:`named tuple`::
```

I'd also like to see an explicit mention there, that the "separate function" referenced there is _not_ provided in the stdlib. Now it just leaves the reader dry on that ("okay, which function???").

On the other hand, RFC 2396 (from 1998) has been obsoleted by [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986) (in 2005), and the latter no longer has the query-params-per-path-segment format in the spec (https://stackoverflow.com/a/6548553/165629).

I wonder how useful it is to reference a spec which is no longer valid and which (probably few people) use. Could the paragraph be lifted altogether?

Then again, it makes perfect sense to explain _why_ there is `urlparse` and `urlsplit`, when/if the only difference is parsing the parameters vs not (I wonder what _is_ the reasoning that remains, it doesn't do real harm to try parse the `?` query parameters when they don't exist).